test/unit/scheduled_publishing_checker_test.rb

```diff
@@ -0,0 +1,105 @@
+    FactoryBot.create(:statistics_announcement, publication: publication, release_date: 1.day.ago)
+
+    test_url = 'http://test.com/published_edition/12'
+    redirect_url = 'http://test.com/announcement/12'
+    Whitehall.url_maker.stubs(:public_document_url).returns(test_url)
+    Whitehall.url_maker.stubs(:statistics_announcement_path).returns(redirect_url)
+
+    setup_server(urls: [test_url], redirects: { redirect_url => test_url })
+
+    ScheduledPublishingChecker.new.check
+
+    assert_equal 1, MissedScheduledPublishing.count
+    assert_equal ScheduledPublishingChecker::SUCCESSFULLY_PUBLISHED, MissedScheduledPublishing.last.status
+  end
+
+  def test_error_when_published_edition_with_redirected_announcement_to_incorrect_path
+    publication = FactoryBot.create(:publication, :statistics, :scheduled, scheduled_publication: 1.day.ago)
+    FactoryBot.create(:statistics_announcement, publication: publication, release_date: 1.day.ago)
+
+    test_url = 'http://test.com/published_edition/12'
+    redirect_url = 'http://test.com/announcement/12'
+    Whitehall.url_maker.stubs(:public_document_url).returns(test_url)
+    Whitehall.url_maker.stubs(:statistics_announcement_path).returns(redirect_url)
+
+    setup_server(urls: [test_url], redirects: { redirect_url => 'http://some.other/path' })
+
+    ScheduledPublishingChecker.new.check
+
+    assert_equal 1, MissedScheduledPublishing.count
+    assert_equal ScheduledPublishingChecker::MISSING_REDIRECT, MissedScheduledPublishing.last.status
+  end
+
+  def test_error_when_published_edition_with_non_redirected_announcement
+    publication = FactoryBot.create(:publication, :statistics, :scheduled, scheduled_publication: 1.day.ago)
+    FactoryBot.create(:statistics_announcement, publication: publication, release_date: 1.day.ago)
+
+    test_url = 'http://test.com/published_edition/12'
+    redirect_url = 'http://test.com/announcement/12'
+    Whitehall.url_maker.stubs(:public_document_url).returns(test_url)
+    Whitehall.url_maker.stubs(:statistics_announcement_path).returns(redirect_url)
+
+    setup_server(urls: [test_url, redirect_url], redirects: [])
+
+    ScheduledPublishingChecker.new.check
+
+    assert_equal 1, MissedScheduledPublishing.count
+    assert_equal ScheduledPublishingChecker::MISSING_REDIRECT, MissedScheduledPublishing.last.status
+  end
+
+  def test_error_when_missing_published_edition
+    FactoryBot.create(:edition, :scheduled, scheduled_publication: 1.day.ago)
+
+    test_url = 'http://test.com/published_edition/12'
+    Whitehall.url_maker.stubs(:public_document_url).returns(test_url)
+
+    setup_server(urls: [], redirects: [])
+
+    ScheduledPublishingChecker.new.check
+
+    assert_equal 1, MissedScheduledPublishing.count
+    assert_equal ScheduledPublishingChecker::MISSING_EDITION, MissedScheduledPublishing.last.status
+  end
+
+  def setup_server(urls:, redirects:)
```

Comment:
Oh, I see now why we're using sinatra - for a fake server to hit in tests.  Could we use webmock instead as it's already in the gemfile?